unscented_kalman_filter/src/ukf_filter.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "ukf_filter.h"
+
+namespace ukfilter {
+//Constructor
+    UKFilter::UKFilter() {}
+
+//##################<Prediction Cycle>#####################################
+    void UKFilter::Prediction(VectorXd &x, MatrixXd &P, double delta_t_) {
+//UKFilter need the argument as state vector x, Covariance Matrix P and timestamp delta_t
+//Basically there are three steps
+//1) Generate Sigma Points with or without augmentation as required
+//2) Predict Sigma Points using nine dimension vehicle model
+//3) Convert the predicted sigma points into predicted mean and covariance
+
+        // Set State space dimension
+        int n_x = x.size();
```

Comment:
It seems the size of state vector will always be 9 even we are using in kinematic model, will that cause an issue?

Reply:
As discussed